.github/workflows/ci.yml

```diff
@@ -87,6 +87,39 @@ jobs:
       - name: Run prettier 💅
         run: npm run lint-prettier-ci
 
+  sonar-scan:
+    permissions:
+      checks: write
+      contents: read
+    needs: [test-ubuntu, test-windows]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 20
+          fetch-tags: false
+
+      - name: Setup workspace ⚙️
+        uses: ./.github/actions/setup-workspace
+        with:
+          os: ubuntu-latest
+          node-version: 20
+
+      - name: Build 🔧
+        run: npm run build
+
+      - name: Run tests with coverage 🧪
+        run: npm run test -- --coverage
+        env:
+          CI: true
+          DISABLE_MOCKED_WARNING: true
+
+      - name: SonarCloud Scan 📊
```

Comment:
Hi @kulshekhar seems like Sonar also offers code coverage report too. Though we would need to disable automatic scan to manually run the action to grab the coverage report. https://community.sonarsource.com/t/how-to-disable-auto-scan/20941

Would you pls help me to disable auto scan and also add `SONAR_TOKEN` to secret envs? Thank you 🙏